core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/WorkspaceResourceMonitor.scala

```diff
@@ -0,0 +1,90 @@
+package org.broadinstitute.dsde.rawls.monitor.workspace
+
+import akka.actor.{Actor, Props}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord.JobType.JobType
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{WorkspaceManagerResourceJobRunner, WorkspaceManagerResourceMonitorRecord}
+import org.broadinstitute.dsde.rawls.dataaccess.{SlickDataSource, WorkspaceManagerResourceMonitorRecordDao}
+import org.broadinstitute.dsde.rawls.monitor.workspace.WorkspaceResourceMonitor._
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+import scala.util.Failure
+
+
+
+object WorkspaceResourceMonitor extends {
+
+  def props(dataSource: SlickDataSource, jobRunners: List[WorkspaceManagerResourceJobRunner])(implicit
+    executionContext: ExecutionContext
+  ): Props = Props(WorkspaceResourceMonitor(dataSource, jobRunners))
+
+  def apply(dataSource: SlickDataSource, jobRunners: List[WorkspaceManagerResourceJobRunner])(implicit
+    executionContext: ExecutionContext
+  ): WorkspaceResourceMonitor =
+    new WorkspaceResourceMonitor(new WorkspaceManagerResourceMonitorRecordDao(dataSource), jobRunners)
+
+  sealed trait WSMJobMonitorMessage
+
+  case object CheckNow extends WSMJobMonitorMessage
+
+  case class CheckDone(creatingCount: Int) extends WSMJobMonitorMessage
+
+  // case class CheckJob(job: WorkspaceManagerResourceMonitorRecord) extends WSMJobMonitorMessage
+
+}
+
+
+class WorkspaceResourceMonitor(jobDao: WorkspaceManagerResourceMonitorRecordDao,
+                               jobRunners: List[WorkspaceManagerResourceJobRunner]
+)(implicit val executionContext: ExecutionContext)
+    extends Actor
+    with LazyLogging {
+
+
+  val registeredRunners: Map[JobType, List[WorkspaceManagerResourceJobRunner]] = jobRunners.groupBy(_.jobType)
```

Comment:
Is it possible to get some test coverage of this file?

Reply:
I've been trying to figure out how to do just that, in a way that is both sane and useful